Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -119,6 +128,14 @@ public class PushDeploymentController : Controller
         {
             using (_tracer.Step("ZipPushDeployViaUrl"))
             {
+                string deploymentId = null;
+                Microsoft.Extensions.Primitives.StringValues idValues;
+
+                if (Request.Headers.TryGetValue(Constants.ScmDeploymentIdHeader, out idValues) && idValues.Count() > 0)
+                {
+                    deploymentId = idValues.ElementAt(0);
+                }
+
```

Comment:
Would be good to move this logic to a helper

Reply:
Will do.

Reply:
Taken care of.